internal/cli/server/server.go

```diff
@@ -160,60 +166,131 @@ func initContextMiddleware(logger logr.Logger) gin.HandlerFunc {
 	}
 }
 
-// authMiddleware authenticates the user either using the session or if one
-// doesn't exist, it authenticates with basic auth.
-func authMiddleware(ctx *gin.Context) {
-	reqCtx := ctx.Request.Context()
-	logger := requestctx.Logger(reqCtx).WithName("AuthMiddleware")
+var oidcProvider *dex.OIDCProvider
 
-	userMap, err := loadUsersMap(ctx)
-	if err != nil {
-		response.Error(ctx, apierrors.InternalError(err))
-		ctx.Abort()
-		return
+// getOIDCProvider returns a lazy constructed OIDC provider
+func getOIDCProvider(ctx context.Context) (*dex.OIDCProvider, error) {
+	// TODO should this "expire" after a while (key rotations, other auth providers)?
```

Comment:
That's true. I still have the feeling that we need to switch the `storage` configuration though. If the Dex installation that we are going to provide is meant to be "production-ready" we should go for the Kubernetes option probably: https://dexidp.io/docs/storage/#configuration

```yaml
storage:
  type: kubernetes
  config:
    inCluster: true
```

We are using the `memory`, and that means that a restart will cause the old tokens to be invalid.

Reply:
The type of storage used by dex has nothing to do with this code though, no? The change you refer to, would happen on the helm-chart.

Reply:
Yes, correct, I've changed it to the Kubernetes type. It was just a self reminder because I guess that the key-rotation and failure that I experienced was due to this.